src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java

```diff
@@ -153,8 +153,15 @@ private CredentialsProvider getApiCredentialsProvider(
             final Secret restApiKey
     ) {
         final CredentialsProvider provider = new BasicCredentialsProvider();
+
+        // Adding check for null or empty restApiKey
+        final String keyTest = restApiKey.getPlainText();
+        if (keyTest == null || keyTest.trim().isEmpty()) {
```

Comment:
Minor: if Apache Commons (e.g. lang3) is in scope, there is a nice `StringUtil.isBlank()` and `StringUtil.isNotBlank()` that applies this logic nicely.